### PR TITLE
[SHIRO-805] Spelling

### PR DESCRIPTION
**Pull request by jsoref**
This PR corrects misspellings identified by the [check-spelling action](https://github.com/marketplace/actions/check-spelling).

The misspellings have been reported at https://github.com/jsoref/shiro/commit/c0e3b03bd1be9f936a8823fd9101593ec62bd8db#commitcomment-45334058

The action reports that the changes in this PR would make it happy: https://github.com/jsoref/shiro/commit/276884c5a738405cd8ef512f14d47b5dd5ce9e31

Note: this PR does not include the action. If you're interested in running a spell check on every PR and push, that can be offered separately.

Following this checklist to help us incorporate your contribution quickly and easily:

 - [x] Make sure there is a [JIRA issue](https://issues.apache.org/jira/browse/SHIRO) filed 
       for the change (usually before you start working on it).  Trivial changes like typos do not 
       require a JIRA issue.  Your pull request should address just this issue, without pulling in other changes.
 - [x] Each commit in the pull request should have a meaningful subject line and body.
 - [ ] Format the pull request title like `[SHIRO-XXX] - Fixes bug in SessionManager`,
       where you replace `SHIRO-XXX` with the appropriate JIRA issue. Best practice
       is to use the JIRA issue title in the pull request title and in the first line of the commit message.
 - [x] Write a pull request description that is detailed enough to understand what the pull request does, how, and why.
 - [ ] Run `mvn clean install apache-rat:check` to make sure basic checks pass. A more thorough check will be performed on your pull request automatically.
 - [ ] If you have a group of commits related to the same change, please squash your commits into one and force push your branch using `git rebase -i`. 
 
Trivial changes like typos do not require a JIRA issue (javadoc, comments...). 
In this case, just format the pull request title like `(DOC) - Add javadoc in SessionManager`.
 
If this is your first contribution, you have to read the [Contribution Guidelines](https://github.com/apache/shiro/blob/master/CONTRIBUTING.md)

If your pull request is about ~20 lines of code you don't need to sign an [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf) 
if you are unsure please ask on the developers list.

To make clear that you license your contribution under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)
 - [x] In any other case, please file an [Apache Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

I expect to squash this PR near the end. But it's much easier for me to address review comments before squashing rather than after, as they generally address individual word families.

If there are files that you don't want changed, please let me know -- dropping them is fairly easy.

### CODE REVIEW

**Comment by bmarwell** [comment]
Thanks!
Can you kindly open a jira issue for this?

Btw, some classes get renamed. If they are not in a test folder, we'd need to let them misspelled the way they are.

File: web/src/main/java/org/apache/shiro/web/filter/mgt/PathMatchingFilterChainResolver.java [hunk 1]
```diff
@@ -101,7 +101,7 @@ public FilterChain getChain(ServletRequest request, ServletResponse response, Fi
 
         String requestURI = getPathWithinApplication(request);
 
-        // in spring web, the requestURI "/resource/menus" ---- "resource/menus/" bose can access the resource
+        // in spring web, the requestURI "/resource/menus" ---- "resource/menus/" base can access the resource
```
Comment (bdemers) [inline web/src/main/java/org/apache/shiro/web/filter/mgt/PathMatchingFilterChainResolver.java:104]:
```suggestion
        // in spring web, the requestURI "/resource/menus" ---- "resource/menus/" both can access the resource
```

File: core/src/main/java/org/apache/shiro/session/Session.java [hunk 2]
```diff
@@ -149,7 +149,7 @@
      * It is common for a {@code Subject} implementation to retain authentication state in the
      * {@code Session}.  If the session
      * is explicitly stopped by application code by calling this method directly, it could clear out any
-     * authentication state that might exist, thereby effectively &quot;unauthenticating&quot; the {@code Subject}.
+     * authentication state that might exist, thereby effectively &quot;deauthenticating&quot; the {@code Subject}.
```
Comment (bdemers) [inline core/src/main/java/org/apache/shiro/session/Session.java:152]: I think the original _might_ be more correct. Though it's possible this line needs to be reworded

Comment (jsoref) [reply core/src/main/java/org/apache/shiro/session/Session.java:152]: Fwiw, this one wasn't automated, I actually read through it and thought about it, but I'm absolutely flexible for all changes / totally willing to drop things.

**Comment by jsoref** [comment]
For the curious, this forced push just folded 8fce45ce into 8fd6e14b.

It sounds like I'm going to split this PR into a couple of pieces, and it's helpful if my commits are reasonably labeled before I do that.

**Comment by jsoref** [comment]
This is now roughly:
* tests
* comments
* docs

The commits I've [dropped](https://github.com/apache/shiro/compare/4491ddc12e2f927a3508902a8c0133023dded827..5211e982aef70d8ee1a2cb48668e4586f06bcbf5) are variable/function/class renames. Note for simplicity, if a code rename includes a comment change, that was included.

File: core/src/main/java/org/apache/shiro/session/Session.java (hunk 2 above)
Comment (jsoref) [reply core/src/main/java/org/apache/shiro/session/Session.java:152]: Fwiw, this seems ok to me. Should I apply it?

**Comment by fpapon** [comment]
@jsoref can you squash the commits please?

**Comment by jsoref** [comment]
(This is just a rebased version before squashing.)

**Comment by jsoref** [comment]
@fpapon ?

File: core/src/main/java/org/apache/shiro/session/Session.java (hunk 2 above)
Comment (bmarwell) [reply core/src/main/java/org/apache/shiro/session/Session.java:152]: Yes, please.

**Comment by bmarwell** [comment]
Maybe just https://github.com/apache/shiro/pull/269#discussion_r547342567 left, then we are good imho.

**Review by fpapon — APPROVED** [review]
LGTM

File: core/src/main/java/org/apache/shiro/authc/pam/ModularRealmAuthenticator.java [hunk 3]
```diff
@@ -100,7 +100,7 @@ public ModularRealmAuthenticator() {
     ============================================*/
 
     /**
-     * Sets all realms used by this Authenticator, providing PAM (Pluggable Authentication Module) configuration.
+     * Set al. realms used by this Authenticator, providing PAM (Pluggable Authentication Module) configuration.
```
Comment (bmarwell) [inline core/src/main/java/org/apache/shiro/authc/pam/ModularRealmAuthenticator.java:103]: Thats a false positive

File: core/src/main/java/org/apache/shiro/session/Session.java [hunk 4]
```diff
@@ -149,7 +149,7 @@
      * It is common for a {@code Subject} implementation to retain authentication state in the
      * {@code Session}.  If the session
      * is explicitly stopped by application code by calling this method directly, it could clear out any
-     * authentication state that might exist, thereby effectively &quot;unauthenticating&quot; the {@code Subject}.
+     * authentication state that might exist, thereby effectively removing the &quot;authenticated&quot; state of the {@code Subject}.
```
Comment (bmarwell) [inline core/src/main/java/org/apache/shiro/session/Session.java:152]: Not 100% sure about this one. I know "to unauthenticate" is not a common word, but it is easier to understand than "removing the >authenticated<-state".

File: core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java [hunk 5]
```diff
@@ -112,7 +112,7 @@ public void testUnSaltedWrongPassword() throws Exception {
         
         Subject.Builder builder = new Subject.Builder(securityManager);
         Subject currentUser = builder.buildSubject();
-        UsernamePasswordToken token = new UsernamePasswordToken(username, "passwrd");
+        UsernamePasswordToken token = new UsernamePasswordToken(username, "password");
```
Comment (bmarwell) [inline core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java:115]: no, test

File: core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java [hunk 6]
```diff
@@ -132,7 +132,7 @@ public void testUnSaltedMultipleRows() throws Exception {
...
```
Comment (bmarwell) [inline core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java:135]: no, test

File: core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java [hunk 7]
```diff
@@ -179,7 +179,7 @@ public void testSaltColumnWrongPassword() throws Exception {
...
```
Comment (bmarwell) [inline core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java:182]: no, test

File: core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java [hunk 8]
```diff
@@ -197,7 +197,7 @@ public void testBase64SaltColumnWrongPassword() throws Exception {
 
         Subject.Builder builder = new Subject.Builder(securityManager);
         Subject currentUser = builder.buildSubject();
-        UsernamePasswordToken token = new UsernamePasswordToken(username, "passwrd");
+        UsernamePasswordToken token = new UsernamePasswordToken(username, "password");
```
Comment (bmarwell) [inline core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java:200]: no, test

File: core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java [hunk 9]
```diff
@@ -228,7 +228,7 @@ public void testExternalWrongPassword() throws Exception {
         
         Subject.Builder builder = new Subject.Builder(securityManager);
         Subject currentUser = builder.buildSubject();
-        UsernamePasswordToken token = new UsernamePasswordToken(username, "passwrd");
+        UsernamePasswordToken token = new UsernamePasswordToken(username, "password");
```
Comment (bmarwell) [inline core/src/test/java/org/apache/shiro/realm/jdbc/JDBCRealmTest.java:231]: no, test

**Review by bmarwell — CHANGES_REQUESTED** [review]
some nits

File: core/src/main/java/org/apache/shiro/authc/pam/ModularRealmAuthenticator.java (hunk 3 above)
Comment (jsoref) [reply core/src/main/java/org/apache/shiro/authc/pam/ModularRealmAuthenticator.java:103]: Technically not a false positive, just a really careless replacement -- oops. Will drop

File: core/src/main/java/org/apache/shiro/session/Session.java (hunk 4 above)
Comment (jsoref) [reply core/src/main/java/org/apache/shiro/session/Session.java:152]: _unauthenticate_ really isn't a word. Conceptually the word would be _deauthenticate_ (but even that isn't in Google Chrome's dictionary -- nor is it in the dictionary I'm using, although I'd be ok w/ it).

Comment (bmarwell) [reply core/src/main/java/org/apache/shiro/session/Session.java:152]: Maybe @bdemers can think of something else?

**Comment by jsoref** [comment]
Fwiw, I'm trying to update this PR, I have everything resolved, but there's quite a bit of additional drift between when I started and now... So, I'm also trying to get my tool to give new happier output.

**Comment by bmarwell** [comment]
> Fwiw, I'm trying to update this PR, I have everything resolved, but there's quite a bit of additional drift between when I started and now... So, I'm also trying to get my tool to give new happier output.

Sorry it took us so long. Shouldn't happen again! :)

**Review by bdemers — APPROVED** [review]

File: core/src/main/java/org/apache/shiro/realm/activedirectory/ActiveDirectoryRealm.java [hunk 10]
```diff
@@ -159,8 +159,8 @@ protected AuthorizationInfo buildAuthorizationInfo(Set<String> roleNames) {
         Set<String> roleNames;
         roleNames = new LinkedHashSet<String>();
 
-        SearchControls searchCtls = new SearchControls();
```
Comment (bmarwell) [inline core/src/main/java/org/apache/shiro/realm/activedirectory/ActiveDirectoryRealm.java:162]: Not strictly necessary, but ok for me.

**Review by bmarwell — APPROVED** [review]

**Comment by fpapon** [comment]
@jsoref thanks for your contribution!

**Comment by sunshineandy** [comment]
view it on GitHub


------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "apache/shiro"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2021年8月4日(星期三) 晚上6:13
***@***.***&gt;;
***@***.***&gt;;
主题:&nbsp;Re: [apache/shiro] [SHIRO-805] Spelling (#269)





 
@jsoref thanks for your contribution!
 
—
You are receiving this because you are subscribed to this thread.
Reply to this email directly, view it on GitHub, or unsubscribe.
Triage notifications on the go with GitHub Mobile for iOS or Android.

**Pull request merged**